src/renderers/abstract.renderer.ts

```diff
@@ -698,6 +698,14 @@ export abstract class Renderer {
       }
     })
 
+    /*
+     * Because of CSP, some ZIM reader environments do not allow inline JS. See issues/2096.
+     */
+    const scripts = Array.from(parsoidDoc.getElementsByTagName('script')) as DominoElement[]
```

Comment:
I guess this will also remove non-inline scripts, i.e. scripts with `src`? These are not blocked by CSP. I mean, I have never seen an essential script in Wikimedia archives, but I'm a little nervous about removing them all. I guess with sufficient testing, we might be able to say that in current Wikimedia ZIMs there are no undesirable effects... And we could modify this if there turned out to be some pages that require scripts for specific features, i.e. a release-and-fix approach...

Reply:
Good catch! However, the way mwoffliner works is that it only takes the `<body>` tag from the returned HTML. It processes other tags to figure out the "JS module dependencies" for the article, but downloads those separately and re-inserts links to the in-ZIM versions of those scripts later.

So while this code technically wipes out all `<script>` tags in the document, in practice any that were not in the `<body>` were going to be thrown out anyways.